test/cpp_extensions/open_registration_extension.cpp

```diff
@@ -137,10 +137,17 @@ void custom_set_backend_meta(const at::Tensor& t) {
 // A dummy storageImpl for our custom device, that secretly uses the CPU
 c10::intrusive_ptr<c10::StorageImpl> make_custom_storage_impl(c10::StorageImpl::use_byte_size_t,
                                                               c10::SymInt size_bytes,
+                                                              c10::DataPtr data_ptr,
                                                               c10::Allocator* allocator,
                                                               bool resizable) {
-  c10::intrusive_ptr<c10::StorageImpl> custom_storage_impl = c10::make_intrusive<c10::StorageImpl>(
+  c10::intrusive_ptr<c10::StorageImpl> custom_storage_impl;
+  if (data_ptr == nullptr){
+    custom_storage_impl = c10::make_intrusive<c10::StorageImpl>(
       c10::StorageImpl::use_byte_size_t(), size_bytes, allocator, resizable);
```

Comment:
This is kind of perplexing, why not just pass in the nullptr and have the constructor deal with it

Reply:
> This is kind of perplexing, why not just pass in the nullptr and have the constructor deal with it

I thought constructor of StorageImpl is a basic function and don't want to modify it at first. Now I allowed  ``data_ptr`` to be nullptr in StorageImpl constructor, could you please take some time to look at the new commits, @ezyang ?